Carousel: Improve test coverage

DylanPiercey opened the pull request:
## Description
This PR adds additional tests for the mobile scrolling version of the carousel and the two utilities `on-scroll-end` and `scroll-transition` associated with mobile scrolling.

While adding these tests I noticed that in the scroll end case the `carousel-update` event was not firing and that fix is included in this PR.

## References
Resolves #309

yomed commented on src/common/test-utils/browser.js:
```diff
@@ -21,7 +21,35 @@ function testOriginalEvent(spy) {
+function simulateScroll(el, to) {
+    const { scrollLeft } = el;
+    const distance = to - scrollLeft;
+    const duration = 100;
```
Was there a specific reason you changed this to 100 from the 10 it was previously? It looks like the failing tests might need higher timeouts to compensate.

DylanPiercey replied:
Yeah I needed a higher duration to be able to cancel the scroll effectively. I'll mess around with the timings.

DylanPiercey commented:
Added another commit to this PR which improves the resiliency of some of the existing carousel related tests by no longer using arbitrary timeouts but hard coded animation frame counts which are reliable. @yomed I also changed the test scroll transition utility to always do the transition over 4 frames instead of 100ms so that its both faster (usually) and can more precisely tested.
